src/ZeroDegreeCalorimeterSiPMonTile_geo.cpp

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2023 Sebouh J. Paul
+
+//==========================================================================
+//  Implementation of SiPM-on-tile Zero-Degree Calorimeter
+//--------------------------------------------------------------------------
+//  Author: Sebouh J. Paul (UCR)
+//==========================================================================
+
+#include "DD4hep/DetFactoryHelper.h"
+#include <XML/Helper.h>
+#include <XML/Layering.h>
+
+using namespace dd4hep;
+
+static Ref_t createDetector(Detector& desc, xml_h handle, SensitiveDetector sens)
+{
+  xml_det_t     detElem = handle;
+  std::string   detName = detElem.nameStr();
+  int           detID   = detElem.id();
+
+  xml_dim_t  dim        = detElem.dimensions();
+  double     width      = dim.x(); // Size along x-axis
+  double     height     = dim.y(); // Size along y-axis
+  double     length     = dim.z(); // Size along z-axis
+
+  xml_dim_t  pos        = detElem.position(); // Position in global coordinates
+
+  Material   air        = desc.material("Air");
+
+  // Defining envelope
+  Box envelope(width / 2.0, height / 2.0, length / 2.0);
+
+  // Defining envelope volume
+  Volume envelopeVol(detName, envelope, air);
+  // Setting envelope attributes
+  envelopeVol.setAttributes(
+    desc,
+    detElem.regionStr(),
+    detElem.limitsStr(),
+    detElem.visStr()
+  );
+
+  PlacedVolume pv;
+
+  double z_distance_traversed = 0.;
+
+  int layer_num = 1;
+
+  // Looping through all the different layer sections
+  for(xml_coll_t c(detElem,_U(layer)); c; ++c)
+  {
+    xml_comp_t x_layer = c;
+    int repeat = x_layer.repeat();
+    double layer_thickness = x_layer.thickness();
+
+    // Looping through the number of repeated layers in each section
+    for(int i = 0; i < repeat; i++)
+    {
+      std::string layer_name = detName + _toString(layer_num, "_layer%d");
+
+      Box layer(width / 2., height / 2., layer_thickness / 2.);
+
+      Volume layer_vol(layer_name, layer, air);
```

Comment:
Ideally, you create the logical layer volume once and place it several times. The flow would be:
- Create logical layer volume
- Loop over slices
  - Create logical slice volume
  - Place logical slice volume inside logical layer volume
- Loop over layers
  - Place logical layer volume in the logical envelope volume
- Place logical envelope volume in the detector's mother volume

Reply:
Dear @veprbl , 
I had patterned this after other detectors in this repository with slices within repeated layers, such as src/InsertCalorimeter_geo.cpp  and src/EndcapCalorimeterWithInsertCutout_geo.cpp, which use similar technology, and I modified the code for this one to have it be the correct shape.  If you can find an example of a detector that does what you mention (ie, uses a single logical layer volume instance that is reused and placed multiple times inside the envelope volume), please let me know.  
Thanks,
Sebouh

Reply:
Thank you for kind pointers. Let's leave this up to a future refactoring then. I will refile my comment as an issue.